lib/ansible/modules/monitoring/zabbix/zabbix_user.py

```diff
@@ -0,0 +1,621 @@
+
+        user_medias = self.convert_user_medias_parameter_types(user_medias)
+
+        user_ids = {}
+
+        request_data = {
+            'userid': zbx_user['userid'],
+            'alias': alias,
+            'name': name,
+            'surname': surname,
+            'usrgrps': user_group_ids,
+            'lang': lang,
+            'theme': theme,
+            'autologin': autologin,
+            'autologout': autologout,
+            'refresh': refresh,
+            'rows_per_page': rows_per_page,
+            'url': url,
+            'type': user_type
+        }
+
+        if override_passwd:
+            request_data['passwd'] = passwd
+
+        # In the case of zabbix 3.2 or less, it is necessary to use updatemedia method to update media.
+        if zbx_api_version <= 32:
+            try:
+                user_ids = self._zapi.user.update(request_data)
+            except Exception as e:
+                self._zapi.logout()
+                self._module.fail_json(msg="Failed to update user %s: %s" % (alias, e))
+
+            try:
+                user_ids = self._zapi.user.updatemedia({
+                    'users': [{'userid': zbx_user['userid']}],
+                    'medias': user_medias
+                })
+            except Exception as e:
+                self._zapi.logout()
+                self._module.fail_json(msg="Failed to update user medias %s: %s" % (alias, e))
+
+        if zbx_api_version >= 34:
+            try:
+                request_data['user_medias'] = user_medias
+                user_ids = self._zapi.user.update(request_data)
+            except Exception as e:
+                self._zapi.logout()
+                self._module.fail_json(msg="Failed to update user %s: %s" % (alias, e))
+
+        return user_ids
+
+    def delete_user(self, zbx_user, alias):
+        user_ids = {}
+        try:
+            user_ids = self._zapi.user.delete([zbx_user['userid']])
+        except Exception as e:
+            self._zapi.logout()
+            self._module.fail_json(msg="Failed to delete user %s: %s" % (alias, e))
+
+        return user_ids
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            server_url=dict(type='str', required=True, aliases=['url']),
+            login_user=dict(type='str', required=True),
+            login_password=dict(type='str', required=True, no_log=True),
+            http_login_user=dict(type='str', required=False, default=None),
+            http_login_password=dict(type='str', required=False, default=None, no_log=True),
+            validate_certs=dict(type='bool', required=False, default=True),
+            alias=dict(type='str', required=True),
+            name=dict(type='str', default=''),
+            surname=dict(type='str', default=''),
+            usrgrps=dict(type='list', required=True),
+            passwd=dict(type='str', required=True, no_log=True),
+            override_passwd=dict(type='bool', required=False, default=False),
```

Comment:
`passwd` parameter is `required` even if `override_passwd` is set to `False`. For me this does not make sense. If I don't want to override the password I don't want to set the `passwd` parameter as well. 
Maybe we can go with `required_one_of` here. What do you think?